extensions/product-recs/bundle/src/main/java/com/adobe/cq/commerce/extensions/recommendations/internal/servlets/AllowedRecTypesDataSourceServlet.java

```diff
@@ -0,0 +1,178 @@
+        List<Resource> allowedRecTypes = new ArrayList<>();
+
+        ResourceResolver resolver = request.getResourceResolver();
+        Resource contentResource = resolver.getResource((String) request.getAttribute(Value.CONTENTPATH_ATTRIBUTE));
+        ContentPolicyManager policyManager = resolver.adaptTo(ContentPolicyManager.class);
+        if (contentResource != null && policyManager != null) {
+            ContentPolicy policy = policyManager.getPolicy(contentResource);
+            if (policy != null) {
+                ValueMap props = policy.getProperties();
+                if (props != null) {
+                    // String[] headingElements = props.get(PN_ALLOWED_HEADING_ELEMENTS, String[].class);
+                    String[] allowedTypeValues = props.get(PN_ALLOWED_TYPES, String[].class);
+                    if (allowedTypeValues != null && allowedTypeValues.length > 0) {
+                        for (String allowedTypeValue : allowedTypeValues) {
+                            RecType recType = RecType.getTypeByValue(allowedTypeValue);
+                            if (recType != null) {
+                                allowedRecTypes.add(new RecTypeResource(recType, resolver));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return allowedRecTypes;
+    }
+
+    /**
+     * Synthetic resource for a product recommendation type that can be used by a Granite form select field.
+     */
+    static class RecTypeResource extends SyntheticResource {
+        public static final String PN_VALUE = "value";
+        public static final String PN_TEXT = "text";
+
+        private final RecType recType;
+        private ValueMap valueMap;
+
+        RecTypeResource(RecType recType, ResourceResolver resourceResolver) {
+            super(resourceResolver, StringUtils.EMPTY, RESOURCE_TYPE_NON_EXISTING);
+            this.recType = recType;
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public <AdapterType> AdapterType adaptTo(Class<AdapterType> type) {
+            if (type == ValueMap.class) {
+                if (valueMap == null) {
+                    initValueMap();
+                }
+                return (AdapterType) valueMap;
+            } else {
+                return super.adaptTo(type);
+            }
+        }
+
+        private void initValueMap() {
+            valueMap = new ValueMapDecorator(new HashMap<String, Object>());
+            valueMap.put(PN_VALUE, getValue());
+            valueMap.put(PN_TEXT, getText());
+        }
+
+        public String getText() {
+            return recType.getText();
+        }
+
+        public String getValue() {
+            return recType.getValue();
+        }
+    }
+
+    /**
+     * Enum with all available product recommendation types.
+     */
+    enum RecType {
+        mostViewed("most-viewed", "Most viewed"),
```

Comment:
please follow the [naming convention of constants in java](https://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html#d5e8254) and make the enum values it all uppercase separated with underscores.

Reply:
Done.